doc/topics/development/releasenotes.rst

```diff
@@ -0,0 +1,71 @@
+.. _releasenotes:
+
+=============
+Release Notes
+=============
+
+You can edit the release notes to highlight a new feature being added
+to a given release. The release notes are templatized with Jinja and
+are generated at release time.
+
+
+.. _edit-release-note:
+
+How do I edit the release notes
+-------------------------------
+
+To edit the release notes you need to look in doc/topics/releases/templates
+for your given release and edit the template. Do not edit the release note
+files in doc/topics/releases/, as this will be written over with the content
+in the template file. For example, if you want to add content to the 3006.0
+release notes you would edit the doc/topics/releases/templates/3006.0.md.template
+file. Do not edit the changelog portion of the template file, since that is
+auto generated with the content generated for the changelog for each release.
+
+
+How to generate the release notes
+---------------------------------
+
+This step is only used when we need to generate the release notes before releasing.
+You should NOT need to run these steps as they are ran in the pipeline, but this
+is documented so you can test your changes to the release notes template.
+
+You will need the `tools` commdan to help run the command. The instructions below
+will detail how to install and use `tools`.
+
+
+Installing `tools`
+..................
+
+.. code-block: bash
+
+    python -m pip install -r requirements/static/ci/py3.10/tools.txt
+
+
+To view the output the release notes will produce before generating them
+you can run `tools` in draft mode:
+
+.. code-block:: bash
+
+    tools changelog update-release-notes --draft
+
+To generate the release notes just remove the `--draft` argument:
+
+.. code-block:: bash
+
+    tools changelog update-release-notes
+
+
+To specify a specific Salt version you add that version as an argument:
+
+.. code-block:: bash
+
+    tools changelog update-release-notes 3006.0
+
+
+To only generate the template for a new release
+
+
+.. code-block:: bash
+
+    tools changelog update-release-notes --template-only
```

Comment:
Are the previous commands all part of a manual step that needs to be done when we're ready for release and the final release notes need to be generated? Does someone run these commands on their local machine and then open up a PR for the final release notes?

Reply:
> Are the previous commands all part of a manual step that needs to be done when we're ready for release and the final release notes need to be generated? Does someone run these commands on their local machine and then open up a PR for the final release notes?

They are not manual steps.These steps are ran in the pipeline at release time but it would be useful to have these steps documented in case someone wanted to test their changes to the template file. 

I included the note in the docs to reiterate that they do not need to run these steps:

"This step is only used when we need to generate the release notes before releasing.
You should NOT need to run these steps as they are ran in the pipeline, but this
is documented so you can test your changes to the release notes template."

> Do you need to do anything differently if you're re-generating previously generated release notes that need to be updated (like if something got missed or a last minute addition)?

You bring up a good point here. @s0undt3ch this is making me think even more that we should add something to pre-commit that auto generates the docs if someone edits the release note template files. We could even ensure it doesn't add the changelog if run from pre-commit.

Reply:
> this is making me think even more that we should add something to pre-commit that auto generates the docs if someone edits the release note template files. We could even ensure it doesn't add the changelog if run from pre-commit.

So, if the template gets modified, what should be updated?

We must remember that we will no longer do docs tags, it either gets in before we release or it doesn't get into the release.

When we stage the release however, we have access to the final rendered documentation, and if something is missing, we commit and stage the release again...

So, what exactly would need to be generate by pre-commit?

Reply:
> We must remember that we will no longer do docs tags, it either gets in before we release or it doesn't get into the release.

This is the change I was not aware of then. If thats the case then I don't see a need to generate the release notes in pre-commit.